lib/routes/account.js

```diff
@@ -18,16 +21,28 @@ module.exports = function (
   db,
   mailer,
   Password,
-  redirectDomain,
-  verifierVersion,
-  isProduction,
-  domain,
-  resendBlackoutPeriod,
+  config,
   customs,
   isPreVerified,
   checkPassword
   ) {
 
+  var openIdExtensions = [
+    new openid.AttributeExchange(
+      {
+        'http://axschema.org/contact/email': 'optional'
+      }
+    )
+  ]
+
+  function isOpenIdProviderAllowed(id) {
+    return config.openIdProviders.some(
+      function (allowed) {
+        return id.indexOf(allowed) === 0
```

Comment:
I think we should do a more thorough parse of the URL and check hostname and root path components explicitly.  We don't want to accidentally allow "https://id.yahoo.com.ihackyou.org" as an IdP because of simple prefix matching.